app/src/main/kotlin/com/hedvig/app/authenticate/LogoutUseCase.kt

```diff
@@ -1,52 +1,39 @@
 package com.hedvig.app.authenticate
 
 import com.apollographql.apollo3.ApolloClient
-import com.hedvig.android.apollo.OperationResult
-import com.hedvig.android.auth.AuthenticationTokenService
+import com.hedvig.android.auth.AuthTokenService
+import com.hedvig.android.core.common.ApplicationScope
+import com.hedvig.android.hanalytics.featureflags.FeatureManager
 import com.hedvig.android.market.MarketManager
 import com.hedvig.app.feature.chat.data.ChatEventStore
 import com.hedvig.app.feature.chat.data.UserRepository
 import com.hedvig.app.service.push.PushTokenManager
 import com.hedvig.app.util.apollo.reconnectSubscriptions
+import com.hedvig.hanalytics.HAnalytics
+import kotlinx.coroutines.launch
 
 class LogoutUseCase(
   private val pushTokenManager: PushTokenManager,
   private val marketManager: MarketManager,
-  private val loginStatusService: LoginStatusService,
   private val apolloClient: ApolloClient,
   private val userRepository: UserRepository,
-  private val authenticationTokenService: AuthenticationTokenService,
+  private val authTokenService: AuthTokenService,
   private val chatEventStore: ChatEventStore,
+  private val featureManager: FeatureManager,
+  private val hAnalytics: HAnalytics,
+  private val applicationScope: ApplicationScope,
 ) {
 
-  sealed class LogoutResult {
-    object Success : LogoutResult()
-    data class Error(val message: String?) : LogoutResult()
-  }
```

Comment:
Logging out should not need a result anymore.
Even in the case where there's no internet connection, we can still clear the tokens locally and then we're good. They will expire soon anyway, so invalidating them can stay as best effort and okay if we don't succeed. Then going back to the marketing activity should be a result of the activity observing the token status and if there are any present or not.

Reply:
Nice!